src/components/sbb-option/sbb-option.tsx

```diff
@@ -0,0 +1,196 @@
+import {
+  Component,
+  ComponentInterface,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  JSX,
+  Listen,
+  Method,
+  Prop,
+  State,
+} from '@stencil/core';
+import { assignId } from '../../global/helpers/assign-id';
+import {
+  createNamedSlotState,
+  queryAndObserveNamedSlotState,
+  queryNamedSlotState,
+} from '../../global/helpers/observe-named-slot-changes';
+import { SbbOptionSelectionChange } from '../sbb-autocomplete/sbb-autocomplete.custom';
+
+let nextId = 0;
+
+/**
+ * @slot unnamed - Use this to provide the option label.
+ * @slot icon - Use this slot to provide an icon. If `icon-name` is set, a sbb-icon will be used.
+ */
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-option.scss',
+  tag: 'sbb-option',
+})
+export class SbbOption implements ComponentInterface {
+  /** Value of the option. */
+  @Prop() public value?: string;
+
+  /**
+   * The icon name we want to use, choose from the small icon variants
+   * from the ui-icons category from here
+   * https://lyne.sbb.ch/tokens/icons/.
+   */
+  @Prop() public iconName?: string;
+
+  /** Whether the icon space is preserved when no icon is set. */
+  @Prop({ reflect: true }) public preserveIconSpace: boolean;
+
+  /** Whether the option is currently active. */
+  @Prop({ reflect: true }) public active?: boolean;
+
+  /** Whether the option is disabled. TBI: missing disabled style, will be implemented with the select component. */
+  @Prop() public disabled?: boolean;
+
+  /** The portion of the highlighted label. */
+  @Prop() public highlightString: string;
+
+  /** Disable the highlight of the label. */
+  @Prop({ reflect: true }) public disableLabelHighlight: boolean;
+
+  /** Emits whenever the menu is closed. */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'option-did-select',
+  })
+  public didSelect: EventEmitter<SbbOptionSelectionChange>;
+
+  /** Emits whenever the menu is closed. */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'option-did-deselect',
+  })
+  public didDeselect: EventEmitter<SbbOptionSelectionChange>;
+
+  /** State of listed named slots, by indicating whether any element for a named slot is defined. */
+  @State() private _namedSlots = createNamedSlotState('icon');
+
+  /** Whether the option is currently selected. */
+  @State() private _selected: boolean;
+
+  @State() private _label: string;
+
+  @Element() private _element: HTMLElement;
+
+  private _optionId = `sbb-option-${++nextId}`;
+  private _labelSlot: HTMLSlotElement;
+
+  @Method()
+  public async select(): Promise<void> {
```

Comment:
Totally agree with you
Initially, I used a property as you're suggesting but then, I realized that I needed to execute some logic on select/deselect (e.g. checking the state and throwing an event).
I couldn't find a clean way to do it since Stencil doesn't allow getters and setters :(

How would you handle this situation?
Would you 'watch' the property and do logic there?

Reply:
I would use a `Watch` as you said. You can take as reference the `sbb-toggle` component or the #1591 PR where Lukas already did something similar by removing the `select()` method from the `sbb-toggle-option`.